src/infrastructure/db/repositories/mongo.repository.ts

```diff
@@ -171,16 +172,30 @@ export class MongoRepository<E> implements DBRepository<E> {
   }
 
   /**
-   * Find a document by email and update password property
+   * Find a document by email and update password, status to ACTIVE and delete expireBlockedAt properties.
    *
    * @param email - Account email address
    * @param password - New password already hashed
+   * @param unlockUser - Update status to ACTIVE and remove expireBlockedAt field.
    * @returns Account document updated
    */
-  async updatePasswordByEmail(email: string, password: string) {
+  async updatePasswordByEmail(
+    email: string,
+    password: string,
+    unlockUser = false,
+  ) {
+    const extraUpdateFields = unlockUser
+      ? {
+          status: ACCOUNT_STATUS.ACTIVE,
+          $unset: { expireBlockedAt: null },
```

Comment:
Here we don't want to remove the `expireBlockedAt` field, we want to set it to null:
```suggestion
          status: ACCOUNT_STATUS.ACTIVE,
          expireBlockedAt: null,
```